azure/services/agentpools/agentpools.go

```diff
@@ -125,6 +128,19 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			},
 		}
 
+		// When autoscaling is set, the count of the nodes differ based on the autoscalar and should not depend on the
+		// count present in machinepool, azuremanagedmachinepool, hence we should not make an update api call based
+		// on difference in count.
+		if profile.EnableAutoScaling != nil && existingProfile.Count != nil {
```

Comment:
I think this needs to be:

`if to.Bool(profile.EnableAutoScaling) && existingProfile.Count != nil {`

adding the `"github.com/Azure/go-autorest/autorest/to"` package to the existing import set in this file

Or this works without the imported library (but is more verbose):

`if profile.EnableAutoScaling != nil && if *profile.EnableAutoScaling && existingProfile.Count != nil {`

Reply:
ack